corehq/apps/locations/dbaccessors.py

```diff
@@ -40,3 +40,25 @@ def get_all_users_by_location(domain, location_id):
         reduce=False,
     )
     return (CouchUser.wrap_correctly(res['doc']) for res in results)
+
+
+def users_have_locations(domain):
+    from corehq.apps.users.models import CouchUser
+    return bool(CouchUser.get_db().view(
+        'locations/users_by_location_id',
+        startkey=[domain],
+        endkey=[domain, {}],
+        reduce=True,
+    ).one())
+
+
+def get_all_users_with_locations(domain):
+    from corehq.apps.users.models import CouchUser
+    results = CouchUser.get_db().view(
```

Comment:
not blocking but I think aspiring to have all new code that iterates over all X in a domain should probably still use `iter_docs` even if we expect X to usually be lower than 100 instances.